File: jce.Front/src/app/@core/data/models/supplier/supplier.ts
import {Good} from '../products/Good';
import {BaseEntity} from '../shared/baseEntity';

class Supplier extends BaseEntity {
  constructor(
    public id: number,
    public  name: string,
    public  description: string,
    public isEnabled: boolean,
    public supplierRef: string,
    public productCount: number,
    public goods: Good[]) {super(); }
}
class FormSupplierModel extends BaseEntity {
  constructor(
    public id: number,
    public  name: string,
    public  description: string,
    public isEnabled: boolean,
    public supplierRef: string,
    public goods: Good[]) {super(); }
}

export {Supplier, FormSupplierModel};

File: jce.Server/jce.Common/Entites/JceDbContext/Supplier.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;
using jce.Common.Core;

namespace jce.Common.Entites.JceDbContext
{
    [Table("Suppliers")]
    public class Supplier : BaseEntity
    {
        [Key]
        public int Id { get; set; }
        [Required]
        [StringLength(255)]
        public string Name { get; set; }
        public string Description { get; set; }
        public bool? IsEnabled { get; set; }
        [Required]
        [StringLength(20)]
        public string SupplierRef { get; set; }

        public ICollection<Product> Products { get; set; }

        public Supplier()
        {
            Products = new Collection<Product>();
        }


    }
}
